src/main/app/reports/ReportsHandler.ts

```diff
@@ -1,65 +1,108 @@
-  private readonly TMP_FOLDER = path.join('/');
+  private readonly store: Store;
+  private readonly reportTimeout = 30 * 60;
 
-  constructor(
+  public constructor(
     private readonly logger: Logger,
     private readonly telemetryClient: TelemetryClient
   ) {
-    // Clear old files every minute
-    setInterval(this.deleteOldReports, 1000 * 60);
+    this.store = this.getStore();
   }
 
-  generate(reportData: User[]): Promise<string | void> {
+  public saveReport(users: User[]): Promise<string | void> {
     const uuid = uuidv4();
-    const fileName = uuid + '.csv';
-    const filePath = path.join(this.TMP_FOLDER, fileName);
-    const fileData = new Parser().parse(reportData);
-
-    return fs.promises.writeFile(filePath, fileData)
+    return this.store
+      .set(uuid, users)
       .then(() => uuid)
       .catch(e => {
-        this.telemetryClient.trackTrace({message: 'Error creating report file: ' + fileName});
         this.logger.error(e);
+        this.telemetryClient.trackTrace(e);
         throw new Error();
       });
   }
 
-  load(fileUUID: string) {
-    const fileName = fileUUID + '.csv';
-    const filePath = path.join(this.TMP_FOLDER, fileName);
-
-    return fs.promises.readFile(filePath)
+  public getReport(reportUUID: string): Promise<string> {
+    return this.store.get(reportUUID)
+      .then((data: string) => JSON.parse(data))
+      .then((data: User[]) => parse(data))
       .catch(e => {
-        this.telemetryClient.trackTrace({message: 'Error loading report file: ' + fileName});
         this.logger.error(e);
-
+        this.telemetryClient.trackTrace(e);
         throw new Error();
       });
   }
 
-  private deleteOldReports = () => {
-    fs.promises.readdir(this.TMP_FOLDER)
-      .then((files: string[]) => {
-        files.map(file => {
-          const filePath = path.join(this.TMP_FOLDER, file);
+  private getStore(): Store {
+    const redisHost: string = config.get('session.redis.host');
+    const redisPort: number = config.get('session.redis.port');
+    const redisPass: string = config.get('session.redis.key');
 
-          fs.promises.stat(filePath)
-            .then((stats: Stats) => {
-              const fileAgeInMinutes = (new Date().getTime() - new Date(stats.mtime).getTime()) / (1000 * 60);
-              if (fileAgeInMinutes >= 1) {
-                fs.promises.unlink(filePath).then(() => console.log('Deleted report file: ' + file));
-              }
-            });
-        });
-      });
+    if (redisHost && redisPass) {
+      this.logger.info('Using Redis Store');
+      return this.getRedisStore(redisHost, redisPort, redisPass);
+    }
+
+    this.logger.info('Using In Memory Store');
+    return this.getInMemoryStore();
+  }
+
+  private getRedisStore(redisHost: string, redisPort: number, redisPass: string): Store {
+    const client = createClient({
+      host: redisHost,
+      password: redisPass,
+      port: redisPort ?? 6380,
+      tls: true
```

Comment:
Is there an option to set a timeout for creating the connection (and maybe retry)?
What we've seen in the Java apps, is:
*  connections are sometimes not able to be established and if the timeout is long (e.g. 15 min), then the app just hangs instead of failing fast
* a connection might get dropped, so we had to add retries on doing an operation in redis